sdks/python/apache_beam/ml/transforms/embeddings/open_ai.py

```diff
@@ -0,0 +1,223 @@
+          a RateLimitError (429) error.
+    """
+  return isinstance(exception, (RateLimitError, APIError))
+
+
+class _OpenAITextEmbeddingHandler(ModelHandler):
+  """
+    Note: Intended for internal use and guarantees no backwards compatibility.
+    """
+
+  def __init__(
+      self,
+      model_name: str,
+      api_key: Optional[str] = None,
+      organization: Optional[str] = None,
+      dimensions: Optional[int] = None,
+      user: Optional[str] = None,
+  ):
+    """Initialize OpenAI embedding handler.
+
+        Args:
+            model_name: Name of the OpenAI embedding model (e.g., "text-embedding-3-small")
+            api_key: OpenAI API key
+            organization: OpenAI organization ID
+            dimensions: Specific embedding dimensions to use (if model supports it)
+            user: End-user identifier for tracking and rate limit calculations
+        """
+    # Store only configuration parameters, not the client
+    self.model_name = model_name
+    self.api_key = api_key
+    self.organization = organization
+    self.dimensions = dimensions
+    self.user = user
+
+    # Configure AdaptiveThrottler and throttling metrics for client-side
+    # throttling behavior.
+    self.throttled_secs = Metrics.counter(
+        OpenAITextEmbeddings, "cumulativeThrottlingSeconds")
+    self.throttler = AdaptiveThrottler(
+        window_ms=1, bucket_ms=1, overload_ratio=2)
+
+  @retry.with_exponential_backoff(
+      num_retries=5, retry_filter=_retry_on_appropriate_openai_error)
+  def get_request(
+      self, text_batch: Sequence[str], model: Any, throttle_delay_secs: int):
+    while self.throttler.throttle_request(time.time() * _MSEC_TO_SEC):
+      LOGGER.info(
+          "Delaying request for %d seconds due to previous failures",
+          throttle_delay_secs)
+      time.sleep(throttle_delay_secs)
+      self.throttled_secs.inc(throttle_delay_secs)
+
+    try:
+      req_time = time.time()
+      kwargs = {
+          "model": self.model_name,
+          "input": text_batch,
+      }
+      if self.dimensions:
+        kwargs["dimensions"] = self.dimensions
+      if self.user:
+        kwargs["user"] = self.user
+
+      response = model.embeddings.create(**kwargs)
+      self.throttler.successful_request(req_time * _MSEC_TO_SEC)
+      return [item.embedding for item in response.data]
+    except RateLimitError as e:
+      LOGGER.warning("Request was rate limited by OpenAI API")
+      raise
+    except Exception as e:
+      LOGGER.error("Unexpected exception raised as part of request: %s", e)
+      raise
+
+  def run_inference(
+      self,
+      batch: Sequence[str],
+      model: Any,
+      inference_args: Optional[dict[str, Any]] = None,
+  ) -> Iterable:
+    # Validate that all elements in the batch are strings
+    for item in batch:
+      if not isinstance(item, str):
+        raise TypeError(
+            f"Embeddings can only be generated on strings. Got {type(item)} instead."
+        )
```

Comment:
No need to explicitly check your input type here, the type checking infrastructure will handle it with the `Sequence[str]` hint